collector/system_collector.go

```diff
@@ -127,17 +236,45 @@ func (sc *systemCollector) collectClusterNodeMetrics() (systemStatusMetrics []sy
 		}
 		systemStatusMetrics = append(systemStatusMetrics, controllerStatusMetric)
 	}
+	return
+}
 
-	for _, m := range clusterNodesStatus.ManagementCluster {
-		managementStatusMetric := systemStatusMetric{
+func (sc *systemCollector) extractManagementNodeMetrics(managementNodes []administration.ManagementNodeAggregateInfo) (systemStatusMetrics []systemStatusMetric) {
+	for _, m := range managementNodes {
+		managementNodeMetric := systemStatusMetric{
 			IPAddress: m.RoleConfig.MgmtPlaneListenAddr.IpAddress,
 			Type:      "management",
 			Status:    0.0,
 		}
 		if strings.ToUpper(m.NodeStatus.MgmtClusterStatus.MgmtClusterStatus) == "CONNECTED" {
-			managementStatusMetric.Status = 1.0
+			managementNodeMetric.Status = 1.0
+		}
+		if len(m.NodeStatusProperties) > 0 {
+			const latestDataIndex = 0
+			prop := m.NodeStatusProperties[latestDataIndex]
+
+			managementNodeMetric.CPUCores = float64(prop.CpuCores)
+
+			managementNodeMetric.LoadAverage = make([]float64, 3)
```

Comment:
improvement:
to remove 0,1,2 index and add meaningful description
we can put load average into struct{timeRange string, value float} or separate into LoadAverageOneMinute
because `Collect` will need to figure out what these three indexes mean

Reply:
this is good, will do follow up PR for this